taskpools/chase_lev_deques.nim

```diff
@@ -111,13 +109,21 @@ proc grow[T](deque: var ChaseLevDeque[T], buf: var ptr Buf[T], top, bottom: int)
   for i in top ..< bottom:
     tmp[][i] = buf[][i]
 
-  buf.prev = deque.garbage
+  buf.prevRetired = deque.garbage
   deque.garbage = buf
   # publish globally
   deque.buf.store(tmp, moRelaxed)
   # publish locally
   swap(buf, tmp)
 
+proc garbageCollect(deque: var ChaseLevDeque) {.inline.} =
+  var node = deque.garbage
```

Comment:
If `garbageCollect` takes the `Buf[T]` instead of the `ChaseLevDeque`, I think the extra `garbage` in `ChaseLevDequeue` would no longer be necessary. We could simply start here at `buf.prevGarbage` instead. `garbageCollect` is only called when no thieves are present, by the owner, so it should be safe imo.